taskRequests/details/script.js

```diff
@@ -251,4 +267,162 @@ const renderTaskRequest = async () => {
   }
 };
 
+function showToast(message, type) {
+  toast.innerHTML = `<div class="message">${message}</div>`;
+  toast.classList.remove('hidden');
+
+  if (type === 'success') {
+    toast.classList.add('success');
+    toast.classList.remove('failure');
+  } else if (type === 'failure') {
+    toast.classList.add('failure');
+    toast.classList.remove('success');
+  }
+
+  setTimeout(() => {
+    toast.classList.add('hidden');
+    toast.innerHTML = '';
+  }, 5000);
+}
+
+async function getAssigneeName() {
+  let userName = '';
+  let res;
+  if (taskRequest.approvedTo) {
+    try {
+      res = await fetch(
+        `${API_BASE_URL}/users/userId/${taskRequest.approvedTo}`,
+      );
+    } catch (error) {
+      console.error(error);
+    }
+    if (res.ok) {
+      const userData = await res.json();
+      userName = userData.user.first_name;
+    }
+  }
+  return userName;
+}
+
+async function renderAssignedTo(userName) {
+  const assignedToText = 'Assigned To: ';
+  const linkOrText = userName.length
+    ? `<a href="https://members.realdevsquad.com/${userName}">${userName}</a>`
+    : 'N/A';
+
+  taskContainer.append(
+    createCustomElement({
+      tagName: 'p',
+      class: 'task__createdBy',
+      id: 'task__createdBy',
+      innerHTML: assignedToText + linkOrText,
+    }),
+  );
+}
+
+const modal = document.getElementById('requestor_details_modal');
+const openModalBtn = document.getElementById('requestor_details_modal_open');
+const closeModal = document.getElementById('requestor_details_modal_close');
+
+closeModal.addEventListener('click', function () {
+  modal.style.display = 'none';
+});
+
+window.addEventListener('click', function (event) {
```

Comment:
why are we adding an event listener on window, this is very bad for the performance

Reply:
Since the user can close the modal by clicking anywhere outside the modal window.

Reply:
for that, we keep the modal inside an overlay that covers the entire screen and apply the click on it
let me know if we need to discuss it

Reply:
Thanks for the suggestion 🙂